main.go

```diff
@@ -50,46 +47,43 @@ func main() {
 		img = adjust(img, *mode, *scale)
 		imaging.Save(img, *output)
 		debug("Image saved to ", *output)
-	} else {
-		// initialize with zero date
-		time_last_success := time.Time{};
+		return
+	}
 
-		online := make(chan int)
-		go wait_online(online)
+	var time_last_success time.Time
 
-		// loop forever and wait for network online events
-		for {
-			// wait for network online message from wpa supplicant
-			<- online
-			debug("Network online")
+	if stat, err := os.Stat(*output); err == nil {
+		time_last_success = stat.ModTime()
+	} else {
+		time_last_success = time.Time{}
+	}
 
-			// FIXME - need to wait a few seconds for DNS?
-			time.Sleep(5 * time.Second)
+	// loop forever and wait for network online events
+	for {
+		if time.Now().Sub(time_last_success).Seconds() < float64(cooldown) {
+			debug("Hit cooldown limit")
+			time.Sleep(time.Duration(cooldown) * time.Second)
+			continue
+		}
 
-			// make sure we don't hammer server every time wifi is turned on
-			if time.Now().Sub(time_last_success).Seconds() > float64(*cooldown) {
+		if *source != "" {
+			img, err = sources[*source]()
+		} else {
+			img, err = custom(*url, *format, *xpath)
+		}
 
-				if *source != "" {
-					img, err = sources[*source]()
-				} else {
-					img, err = custom(*url, *format, *xpath)
-				}
+		if err != nil {
+			fmt.Println(err)
+			continue
+		}
 
-				if err == nil {
-					time_last_success = time.Now()
-				} else {
-					fmt.Println(err)
-					continue
-				}
-			} else {
-				debug("Hit cooldown limit")
-				continue
-			}
+		img = adjust(img, *mode, *scale)
+		imaging.Save(img, *output)
+		debug("Image saved to ", *output)
+		time_last_success = time.Now()
 
-			// img = adjust(img, *top, *left, *right, *bottom)
-			img = adjust(img, *mode, *scale)
-			imaging.Save(img, *output)
-			debug("Image saved to ", *output)
-		}
+		wait_for := time.Duration(cooldown) * time.Second
+		debug(fmt.Sprintf("Sleeping for %v", wait_for))
+		time.Sleep(wait_for)
 	}
```

Comment:
If I'm understanding these changes correctly, once the program is started it will try to download the image source every 3600s until the device is powered down?

What if the device is only briefly connected to the internet, or the user keeps the device suspended for all but a few minutes in the day?

Reply:
bump

Reply:
hmm, that's a good point.  Should probably have the systemd unit do the scheduling since it understands those events and have this just check and exit.

had mentioned in the description that this is probably pointless, as the new remarkable os version requires a device restart to pickup on sleep screen changes, unless i missed something.